lib/desktop/session.rb

```diff
@@ -144,23 +149,33 @@ def clean
     end
 
     def geometry
-      return 'UNKNOWN' unless local?
-      xrandr.readlines[0]
-            .slice(/current (.*),/, 1)
-            .gsub(/\s/, '')
+      begin
+        xrandr.first
+              .slice(/current (.*),/, 1)
+              .gsub(/\s/, '')
+      rescue
+        'UNKNOWN'
+      end
     end
 
     def resize(geometry)
-      raise "cannot resize a remote desktop session" unless local?
       raise "cannot resize the #{type.name} desktop type" unless type.resizable?
       raise "invalid geometry for the #{type.name} desktop type" unless valid_geometry?(geometry)
-      IO.popen(
-        {
-          "flight_DESKTOP_geometry" => geometry,
-          "DISPLAY" => ":#{display}",
-        },
-        ["#{Dir.home}/.local/share/flight/desktop/bin/flight-desktop_geometry.sh"],
-        )
+      check_for_timeout do
+        CommandUtils.with_clean_env do
+          pio = IO.popen(
+            {
+              "flight_DESKTOP_geometry" => geometry,
+              "DISPLAY" => display_full,
+            },
+            ["#{Dir.home}/.local/share/flight/desktop/bin/flight-desktop_geometry.sh"],
+            )
+          sleep(0.1) # needed so the process doesn't get killed before it finishes
+        ensure
+          Process.kill 9, pio.pid
```

Comment:
Sending `SIGKILL` to a process is a bit of a hard hammer.  Usually we'd prefer to send `SIGTERM`/`15`.  That provides the process with an opportunity to clean up that `SIGKILL` doesn't.

Can you change this to:

```
Process.kill 15, pio.pid
sleep 0.1
Process.kill 9, pio.pid
```